.github/workflows/publish-backend-plugin-manager.yaml

```diff
@@ -0,0 +1,145 @@
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > .npmrc
+          if [ "_$(npm dist-tag ls $SCOPE/backend-plugin-manager | grep -e ^$COMMIT_ID || echo '')" != "_" ]; then ALREADY_PUSHED="true"; else ALREADY_PUSHED="false"; fi
+          echo "ALREADY_PUSHED=$ALREADY_PUSHED" >> "$GITHUB_OUTPUT"
+          echo "GH Output: " && cat $GITHUB_OUTPUT
+
+      - name: Get the version of the backend-plugin-manager package
+        id: get_package_version
+        run: |
+          echo "PACKAGE_VERSION=$(node -p "require('./packages/backend-plugin-manager/package.json').version")" >> "$GITHUB_OUTPUT"
+          echo "GH Output: " && cat $GITHUB_OUTPUT
+
+      - name: Get the last version of the backend-plugin-manager package pushed to the NPMJS repository
+        id: get_last_pushed_version
+        run: |
+          echo "LAST_PUSHED_VERSION=$(npm view $SCOPE/backend-plugin-manager version)" >> "$GITHUB_OUTPUT"
+          echo "GH Output: " && cat $GITHUB_OUTPUT
+
+      - name: Get new version to push
+        id: get_new_version_to_push
+        if: steps.test_commit_already_published.outputs.ALREADY_PUSHED == 'false'
+        env:
+          COMMIT_VERSION: ${{ steps.get_package_version.outputs.PACKAGE_VERSION }}
+          LAST_PUSHED_VERSION: ${{ steps.get_last_pushed_version.outputs.LAST_PUSHED_VERSION }}
+        run: |
+          COMMIT_MAJOR=$(echo $COMMIT_VERSION | cut -d. -f1)
+          COMMIT_MINOR=$(echo $COMMIT_VERSION | cut -d. -f2)
+          COMMIT_PATCH=$(echo $COMMIT_VERSION | cut -d. -f3 | cut -d- -f1)
+
+          if [ "$COMMIT_MAJOR.$COMMIT_MINOR.$COMMIT_PATCH" == "$COMMIT_VERSION" ]; then
+              COMMIT_PRERELEASE=""
+          else
+              COMMIT_PRERELEASE=$(echo $COMMIT_VERSION | cut -d- -f2)
+          fi
+
+          if [ -n "$COMMIT_PRERELEASE" ]; then
+              COMMIT_PATCH=$((COMMIT_PATCH-1))
+          fi
+
+          PATCH=$((COMMIT_PATCH+1))
+          if [ -z "$LAST_PUSHED_VERSION" ]; then
+              LAST_PUSHED_VERSION=$COMMIT_MAJOR.$COMMIT_MINOR.$PATCH-janus.0
+          fi
+
+          LAST_PUSHED_MAJOR=$(echo $LAST_PUSHED_VERSION | cut -d. -f1)
+          LAST_PUSHED_MINOR=$(echo $LAST_PUSHED_VERSION | cut -d. -f2)
+          LAST_PUSHED_PATCH=$(echo $LAST_PUSHED_VERSION | cut -d. -f3 | cut -d- -f1)
+          LAST_PUSHED_PRERELEASE_WITHOUT_NUMBER=$(echo $LAST_PUSHED_VERSION | cut -d- -f2 | cut -d. -f1)
+          LAST_PUSHED_PRERELEASE_NUMBER=$(echo $LAST_PUSHED_VERSION | cut -d- -f2 | cut -d. -f2)
+
+          if [ "$COMMIT_MAJOR.$COMMIT_MINOR.$PATCH" == "$LAST_PUSHED_MAJOR.$LAST_PUSHED_MINOR.$LAST_PUSHED_PATCH" ]; then
+              PRERELEASE="janus.$((LAST_PUSHED_PRERELEASE_NUMBER+1))"
+          else
+              PRERELEASE="janus.0"
+          fi
+          echo "VERSION=$COMMIT_MAJOR.$COMMIT_MINOR.$PATCH-$PRERELEASE" >> "$GITHUB_OUTPUT"
+          echo "GH Output: " && cat $GITHUB_OUTPUT
```

Comment:
Would you mind documenting this code a bit? It's a bit complicated and confusing. 🙂